src/integer_mod_q/mat_polynomial_ring_zq/get.rs

```diff
@@ -318,3 +355,53 @@ mod test_mod {
         );
     }
 }
+
+#[cfg(test)]
+mod test_collect_entries {
+    use flint_sys::fmpz_poly::fmpz_poly_set;
+
+    use crate::integer::{MatPolyOverZ, PolyOverZ};
+    use crate::integer_mod_q::{MatPolynomialRingZq, ModulusPolynomialRingZq};
+    use std::str::FromStr;
+
+    const BITPRIME64: u64 = 18446744073709551557;
+
+    #[test]
+    fn all_entries_collected() {
+        let modulus =
+            ModulusPolynomialRingZq::from_str(&format!("4  1 0 0 1 mod {}", BITPRIME64)).unwrap();
+        let poly_mat1 = MatPolyOverZ::from_str(&format!(
+            "[[4  -1 0 3 1, 1  {}],[2  1 2, 3  {} 1 1]]",
+            i64::MAX,
+            i64::MIN,
+        ))
+        .unwrap();
+        let poly_ring_mat1 = MatPolynomialRingZq::from((&poly_mat1, &modulus));
+        let poly_mat2 = MatPolyOverZ::from_str("[[1  42, 2  1 17]]").unwrap();
+        let poly_ring_mat2 = MatPolynomialRingZq::from((&poly_mat2, &modulus));
+
+        let entries_1 = poly_ring_mat1.collect_entries();
+        let entries_2 = poly_ring_mat2.collect_entries();
+
+        let mut entry1 = PolyOverZ::default();
+        let mut entry2 = entry1.clone();
+        let mut entry3 = entry1.clone();
+
+        unsafe { fmpz_poly_set(&mut entry1.poly, &entries_1[1]) }
+        unsafe { fmpz_poly_set(&mut entry2.poly, &entries_1[3]) }
+        unsafe { fmpz_poly_set(&mut entry3.poly, &entries_2[0]) }
+
+        assert_eq!(entries_1.len(), 4);
+        assert_eq!(
+            PolyOverZ::from_str(&format!("1  {}", i64::MAX)).unwrap(),
+            entry1
+        );
+        assert_eq!(
+            PolyOverZ::from_str(&format!("3  {} 1 1", i64::MAX - 58)).unwrap(),
```

Comment:
Use `u64::MAX - 58` if it should be a valid prime number. `i64::MAX - 58` isn't prime: 9223372036854775749 = 3×8447×363970326224489 (3 distinct prime factors)

Reply:
It is not needed to be prime